### PR TITLE
HLREST: Add Clear Roles Cache API

### PR DESCRIPTION
**Pull request by bleskes**
Add's support to the [Clear Roles Cache API](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-clear-role-cache.html) to the High Level Rest Client.

Reviews should note the addition of the NodesResponseHeader utility class (I'm not sure it's in the right place) and also the fact this specific API has a weird response structure: 

```
{
   _nodes: {
       total: 2,
       successful: 2
       failed: 0
   },
  nodes: {
       "id": {
          ...
        }
  }
}
```

I chose to the response object I added to have a `getHeader()` method to return the content of the `_nodes` section and a `getNodes()` method to return the node info (mimicking the response object in ES core). We can also choose to fold the header into the top level response class and expose total, successful, failed and failure directly (at the expense of the class not looking like the JSON)/

Relates to #29827

### CODE REVIEW

**Review by jaymode — CHANGES_REQUESTED** [review]
I left some small comments. It looks good and you beat me to getting my PR up with my own version of `NodesResponseHeader` that is very similar (I did it for the clear realm cache api)

**Review by hub-cap — COMMENTED** [review]
+1 to jays comments, good work so far.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/NodesResponseHeader.java
Comment (jaymode) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/NodesResponseHeader.java]: This would be used by security, watcher, and a few cluster APIs (cluster stats, nodes info, and nodes stats).

**Comment by jaymode** [comment]
@hub-cap I picked this one up from @bleskes so I can unblock myself for clear realm caches. When you get a chance, can you please take a look?

**Review by hub-cap — APPROVED** [review]
I see updates to the Doc test, but i dont see any accompanying asciidoc changes, just making sure these are done properly. Other than that, looks good.

**Comment by bleskes** [comment]
@jaymode sorry. I didn't realize you were blocked on this one. Thanks for picking it up.

**Comment by jaymode** [comment]
@hub-cap good catch on the docs. I just added them

**Review by jaymode — APPROVED** [review]

**Pull request merged**
